src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -1106,20 +1104,10 @@ public IMessage handleColonyBuildingViewMessage(final BlockPos buildingId, @NotN
     }
 
     @Override
-    public void handleColonyFieldViewMessage(final @NotNull FieldRegistries.FieldEntry type, final @NonNull BlockPos position, @NotNull final FriendlyByteBuf buf)
+    public void handleColonyFieldViewUpdateMessage(final Set<IField> fields)
     {
-        final IField fieldView = type.produceField(position);
-        fieldView.deserialize(buf);
-        fields.remove(fieldView);
-        fields.add(fieldView);
-    }
-
-    @Override
-    public void handleColonyRemoveFieldViewMessage(final @NotNull FieldRegistries.FieldEntry type, final @NonNull BlockPos position, @NotNull final FriendlyByteBuf buf)
-    {
-        final IField field = type.produceField(position);
-        field.deserialize(buf);
-        fields.remove(field);
+        this.fields.clear();
+        this.fields.addAll(fields);
```

Comment:
While you got a UI open for a field, that should be at least temporarily cached there

Reply:
In all buildings at least, in the window constructor we query the building and then operate on it. Else, we have to constantly query the list, go through all items in the list and get the correct building. Every frame.

Reply:
I have to do so anyway because the field list updates on each frame anyway, and there's no way to even stop that anyway.

If I then cache it I have to constantly request the new state back from the colony view anyway to get the new active state.
So instead of caching it just immediately asks it directly from the colony view.

Reply:
So each field, goes every frame to the colonyview and asks it for the field list and sorts out their own instance as a basis for truth?

Reply:
we're doing way more expensive stuff on other UIs, this is fine xD

Reply:
Seems odd, especially as we do it differently everywhere else. If we rely in some areas on the instance to remain "fresh" and do it like this in others this invites future bugs to happen.

Reply:
I actually got more issues back then with having several sources of truth. Previously assigned fields were stored within the fields module for example. But I got so many sync related issues I got completely tired of it, hence why I moved every instance of fields to the structure manager and colony view, and everything must be obtained through there.

Reply:
No, I agree, the previous tileEntity system was a mess. I am talking about handling how it is also handled with buildings and colonyviews.